internal/controller/git/git.go

```diff
@@ -30,6 +30,17 @@ type RepoCredentials struct {
 	Password string `json:"password,omitempty"`
 }
 
+// CommitUser represents the user for committing changes to a git repository.
+type CommitUser struct {
```

Comment:
I think this should probably be called `Author` (or `CommitAuthor`). This pairs better with the method you introduced, and avoids using new terminology for something which has already a well established name within the domain (Git).

Reply:
Thinking this over once more... The terminology of Git is three fold.

You can have a "user", which can either be;

1. the author of a patch.
2. the committer of a patch.
3. the author of an annotated tag.

On a desktop computer, this is configured as `git config user.name | git config user.email`. Given this, I would just call the struct `User` or `GitUser`, which makes it suitable to be used for any of the above (and beyond just commits).

Reply:
I will go with `User`, as the package is `git`, the usage will be `git.User`. Thanks for the suggestion!

Reply:
[refactor(controller): rename commit user to user](https://github.com/akuity/kargo/pull/1843/commits/01f9006f35b3f7d17209d55c11fa343a8c047535)